### PR TITLE
Feature/css prefixer

### PR DESCRIPTION
**Pull request by elibosley**
# PR for No Issue .

## PR Status
**Ready**

## Ready for PR?
- [x] I have updated the CHANGELOG with an entry including a description, a link to my profile, and a link to the issue ticket. This change is filed under an Enhancement or Bug Fix, is shown under the relevant release tag name, and is included in my PR branch.
- [x] I have run `lein run-lint` to check for linting errors.

## Changes Proposed:
- Adds autoprefixer which is run before build and when writing code to watch CSS files

@emilyseibert @oconn @dehli

### CODE REVIEW

**Comment by elibosley** [comment]
This isn't done, still need to update the changelog!

**Comment by dehli** [comment]
Could you resolve the conflicts (I think I have some suggestions that you've already fixed)?

File: project.clj [hunk 1]
```diff
@@ -105,9 +107,18 @@
...
-              "deploy-dev"  ["do"  "clean," "run-lint" ["cljsbuild" "once" "deploy-dev"]  ["less" "once"]]
-              "deploy-qa"   ["do"  "clean," "run-lint" ["cljsbuild" "once" "deploy-qa"]   ["less" "once"]]
-              "deploy-prod" ["do"  "clean," "run-lint" ["cljsbuild" "once" "deploy-prod"] ["less" "once"]]})
+    :aliases {"install-postcss" [["shell" "lein" "npm" "install" "postcss-cli" "autoprefixer"]]
+              "postcss-prefixer-w" [["shell" "postcss" "resources/temp/css/*.css" "resources/temp/css/**/*.css" "resources/temp/css/**/**/*.css"
+                              "--use autoprefixer" "-d" "resources/public/css" "-w"]]
+              "postcss-prefixer" [["shell" "postcss" "resources/temp/css/*.css" "resources/temp/css/**/*.css" "resources/temp/css/**/**/*.css"
+                              "--use autoprefixer" "-d" "resources/public/css"]]
+              "run-local"   ["do"
+                             ["install-postcss"]
```
Comment (dehli) [inline project.clj]: Would this work without the brackets?

File: project.clj [hunk 2]
```diff
@@ -105,9 +107,18 @@
...
+                             ["pdo" "clean," ["figwheel" "local"] ["less" "auto"] "postcss-prefixer-w"]]
+              "run-dev"     ["do"
+                             ["install-postcss"]
```
Comment (dehli) [inline project.clj]: Same here?

File: src/cljs/wombats_web_client/events/games.cljs [hunk 3]
```diff
@@ -113,10 +113,10 @@
     #(print "error on get all closed games"))))
 
 (defn get-all-games []
-  (get-open-games)
-  (get-my-open-games)
-  (get-closed-games)
-  (get-my-closed-games))
+  (get-open-games 0)
```
Comment (dehli) [inline src/cljs/wombats_web_client/events/games.cljs]: This should still be 0. Must be a result from the conflicts.

File: src/cljs/wombats_web_client/events/games.cljs [hunk 4]
```diff
@@ -150,9 +150,9 @@
 (re-frame/reg-fx
  :get-open-games
  (fn [_]
-   (get-open-games)))
+   (get-open-games 0)))
```
Comment (dehli) [inline src/cljs/wombats_web_client/events/games.cljs]: Should still be a 0

File: src/cljs/wombats_web_client/events/games.cljs [hunk 5]
```diff
@@ -150,9 +150,9 @@
...
 
 (re-frame/reg-fx
  :get-closed-games
  (fn [_]
-   (get-closed-games)))
+   (get-closed-games 0)))
```
Comment (dehli) [inline src/cljs/wombats_web_client/events/games.cljs]: - 0

File: src/cljs/wombats_web_client/panels/games.cljs [hunk 6]
```diff
@@ -27,6 +27,13 @@
 ;; Helpers
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+<<<<<<< HEAD
```
Comment (dehli) [inline src/cljs/wombats_web_client/panels/games.cljs]: This should be removed.

File: src/cljs/wombats_web_client/panels/games.cljs [hunk 7]
```diff
@@ -143,9 +154,17 @@
      (and (not closed) (not mine))
      empty-open-page)])
 
+<<<<<<< HEAD
```
Comment (dehli) [inline src/cljs/wombats_web_client/panels/games.cljs]: This file wasn't resolved properly.

**Comment by elibosley** [comment]
@dehli This should be good to be re-reviewed!

File: .gitignore [hunk 8]
```diff
@@ -1,7 +1,9 @@
 /target
 /resources/public/js
 /resources/public/css
+/resources/temp/*
 *.log
 .DS_Store
 node_modules/
 out/
+package-lock.json
```
Comment (dehli) [inline .gitignore]: We actually don't want to `.gitignore` this. It should be checked in so we can ensure that the correct versions of `npm` packages are installed.

https://docs.npmjs.com/files/package-lock.json

File: project.clj [hunk 9]
```diff
@@ -105,9 +107,20 @@
                     :output-dir    "resources/public/js/compiled/test/out"
                     :optimizations :none}}]}
 
-    :aliases {"run-local"   ["pdo" "clean," ["figwheel" "local"] ["less" "auto"]]
-              "run-dev"     ["pdo" "clean," ["figwheel" "dev"]   ["less" "auto"]]
-              "run-lint"    ["pdo" "bikeshed" ["kibit" "src/cljs/wombats_web_client/"]]
-              "deploy-dev"  ["do"  "clean," "run-lint" ["cljsbuild" "once" "deploy-dev"]  ["less" "once"]]
-              "deploy-qa"   ["do"  "clean," "run-lint" ["cljsbuild" "once" "deploy-qa"]   ["less" "once"]]
-              "deploy-prod" ["do"  "clean," "run-lint" ["cljsbuild" "once" "deploy-prod"] ["less" "once"]]})
+    :aliases {"install-postcss" [["shell" "lein" "npm" "install" "postcss-cli" "autoprefixer"]]
```
Comment (dehli) [inline project.clj]: You should have a `package.json` file that specifies the files that need to be installed into this repository. Then you should just need to execute `npm install`. That way if you need to update the `npm` dependencies, you just have to update the `.json` file.

File: project.clj [hunk 10]
```diff
@@ -105,9 +107,20 @@
...
+              "postcss-prefixer-w" [["shell" "postcss" "resources/temp/css/*.css" "resources/temp/css/**/*.css" "resources/temp/css/**/**/*.css"
```
Comment (dehli) [inline project.clj]: I think doing resources/temp/css/**/*.css is good enough b/c I believe it covers any arbitrary depth. I could be wrong though.

File: project.clj [hunk 11]
```diff
@@ -105,9 +107,20 @@
...
+                              "--use autoprefixer" "-d" "resources/public/css" "-w"]]
+              "postcss-prefixer" [["shell" "postcss" "resources/temp/css/*.css" "resources/temp/css/**/*.css" "resources/temp/css/**/**/*.css"
```
Comment (dehli) [inline project.clj]: Do these need to be nested arrays? Would it work if it was just:

`["shell" "postcss"... ]`

File: project.clj (hunk 10 above)
Comment (elibosley) [inline project.clj]: It actually didn't as far as I could tell :(

**Comment by elibosley** [comment]
@dehli Alright, I've fixed those issues - hopefully this looks good now!

**Comment by elibosley** [comment]
@dehli What did you mean by update the issue description?

**Pull request merged**